src/Nethermind/Nethermind.Blockchain/FullPruning/FullPruner.cs

```diff
@@ -171,7 +172,11 @@ protected virtual void RunPruning(IPruningContext pruning, Keccak statRoot)
             {
                 pruning.MarkStart();
                 using CopyTreeVisitor copyTreeVisitor = new(pruning, _logManager);
-                VisitingOptions visitingOptions = new() { MaxDegreeOfParallelism = _pruningConfig.FullPruningMaxDegreeOfParallelism };
+                VisitingOptions visitingOptions = new()
+                {
+                    MaxDegreeOfParallelism = _pruningConfig.FullPruningMaxDegreeOfParallelism,
+                    FullScanMemoryBudget = ((long)_pruningConfig.FullPruningMemoryBudgetMb).MiB(),
```

Comment:
This going to be x 1024*1024 too high? If it comes in as MB already then is multiplied by `MiB`